server.js

```diff
@@ -1,35 +1,132 @@
 import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
+import dotenv from "dotenv";
+import topMusicData from "./data/top-music.json";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+const Song = mongoose.model("Song", {
+  "id": Number,
+  "trackName": String,
+  "artistName": String,
+  "genre": String,
+  "bpm": Number,
+  "energy": Number,
+  "danceability": Number,
+  "loudness": Number,
+  "liveness": Number,
+  "valence": Number,
+  "length": Number,
+  "acousticness": Number,
+  "speechiness": Number,
+  "popularity": Number
+} 
+)
+
+if(process.env.RESET_DB) {
+  const resetDataBase = async () => {
+    await Song.deleteMany();
+    topMusicData.forEach(singleSong => {
+      const newSong = new Song(singleSong);
+      newSong.save();
+    })
+  }
+  resetDataBase()
+}
+
 const port = process.env.PORT || 8080;
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+// endpoints 
 app.get("/", (req, res) => {
```

Comment:
This endpoint is not working. However, I think that you can use npm listEndpoints if you want to list the endpoints.